ably/options.go

```diff
@@ -292,56 +338,87 @@ func (opts *ClientOptions) disconnectedRetryTimeout() time.Duration {
 	return defaultOptions.DisconnectedRetryTimeout
 }
 
-func (opts *ClientOptions) restURL() string {
+func (opts *ClientOptions) getRestHost() string {
 	host := opts.RestHost
-	if host == "" {
-		host = defaultOptions.RestHost
-		if opts.Environment == "production" {
-			opts.Environment = ""
-		}
-		if opts.Environment != "" {
-			host = opts.Environment + "-" + host
-		}
+	defaultHost := defaultOptions.RestHost
+	if empty(host) {
+		host = defaultHost
 	}
-	if opts.NoTLS {
-		port := opts.Port
-		if port == 0 {
-			port = 80
-		}
-		return "http://" + net.JoinHostPort(host, strconv.FormatInt(int64(port), 10))
-	} else {
-		port := opts.TLSPort
-		if port == 0 {
-			port = 443
-		}
-		return "https://" + net.JoinHostPort(host, strconv.FormatInt(int64(port), 10))
+	if host == defaultHost && !opts.isProductionEnvironment() {
+		host = opts.Environment + "-" + host
 	}
+	return host
```

Comment:
I feel like this would be clearer by dealing with the more specific cases first (rest host specified, environment specified) and then falling through to the more general case (default host), which also makes it easier to verify against [RSC11](https://www.ably.io/documentation/client-lib-development-guide/features/#RSC11):

```go
if opts.RestHost != "" {
	return opts.RestHost
}
if opts.Environment != "" && !opts.isProductionEnvironment() {
	return opts.Environment + "-" + defaultOptions.RestHost
}
return defaultOptions.RestHost
```